src/commonTest/kotlin/com/charleskorn/kaml/YamlScalarTest.kt

```diff
@@ -174,7 +179,16 @@ class YamlScalarTest : FlatFunSpec({
                         val result = scalar.toDouble()
 
                         test("converts it to the expected double") {
-                            result shouldBe expectedResult
+                            if (expectedResult.isNaN()) {
+                                // this is not correct to compare NaNs
+                                // instead isNaN method should be invoked
+                                // to check whether the value is NaN or not
+                                withClue("$result should be NaN") {
+                                    result.isNaN() shouldBe true
+                                }
```

Comment:
Can we instead use the dedicated `shouldBeNaN` matcher?

Reply:
I didn't know there was a function like this. Thanks!